src/lib/temple/beacon.test.ts

```diff
@@ -0,0 +1,281 @@
+    Buffer.from([109, 111, 99, 107, 32, 101, 100, 112, 107, 115, 105, 103])
+  ),
+  crypto_sign_ed25519_sk_to_curve25519: jest.fn(() => 'mock edsksig'),
+  crypto_kx_client_session_keys: jest.fn(() => ({
+    sharedRx: new Uint8Array(),
+    sharedTx: new Uint8Array()
+  })),
+  crypto_kx_server_session_keys: jest.fn(() => ({
+    sharedRx: new Uint8Array(),
+    sharedTx: new Uint8Array()
+  })),
+  crypto_secretbox_open_easy: jest.fn(() => 'mock secretbox'),
+  crypto_secretbox_easy: jest.fn(() => 'mock secretbox easy'),
+  randombytes_buf: jest.fn(() => 'mock randombytes'),
+  crypto_box_seal: jest.fn(() => Buffer.from('mock cryptobox seal'))
+}));
+
+const MOCK_PUBLIC_KEY = '444e1f4ab90c304a5ac003d367747aab63815f583ff2330ce159d12c1ecceba1';
+
+describe('Beacon', () => {
+  const MOCK_ORIGINAL_KEY = 'something';
+  const MOCK_MODIFIED_KEY = 'beacon_something_pubkey';
+  const MOCK_ORIGINAL_VALUE = 'somevalue';
+  const MOCK_STORAGE_OBJECT = { [MOCK_MODIFIED_KEY]: MOCK_ORIGINAL_VALUE };
+  const SAMPLE_PAYLOAD = 'hello, world!';
+
+  beforeEach(() => {
+    mockBrowserStorageLocal.set.mockClear();
+    mockBrowserStorageLocal.remove.mockClear();
+  });
+  describe('toPubKeyStorageKey', () => {
+    it('Format public key to storage key', () => {
+      const storageKey = toPubKeyStorageKey('myAwesomeDappKey');
+      expect(storageKey).toBe('beacon_myAwesomeDappKey_pubkey');
+    });
+  });
+  describe('saveDAppPublicKey', () => {
+    it('called with correct arguments', async () => {
+      await saveDAppPublicKey(MOCK_ORIGINAL_KEY, MOCK_ORIGINAL_VALUE);
+      expect(mockBrowserStorageLocal.set).toBeCalledWith(MOCK_STORAGE_OBJECT);
+    });
+  });
+  describe('removeDAppPublicKey', () => {
+    it('called with correct data', async () => {
+      await removeDAppPublicKey(MOCK_ORIGINAL_KEY);
+      expect(mockBrowserStorageLocal.remove).toBeCalledWith([MOCK_MODIFIED_KEY]);
+    });
+  });
+  describe('getDAppPublicKey', () => {
+    it('called with correct arguments', async () => {
+      await getDAppPublicKey(MOCK_ORIGINAL_KEY);
+      expect(mockBrowserStorageLocal.get).toBeCalledWith([MOCK_MODIFIED_KEY]);
+    });
+  });
+  describe('fromHex', () => {
+    it('Decode from Hex is working', async () => {
+      const bufferMock = Buffer.from(SAMPLE_PAYLOAD, 'utf8').toString('hex');
+      const termMock = fromHex(bufferMock);
+      expect(typeof termMock).toBe('object');
+    });
+  });
+  describe('toHex', () => {
+    it('Encode to Hex is working', async () => {
+      const termMock = Buffer.from(SAMPLE_PAYLOAD, 'utf8');
+      const bufferMock = toHex(termMock);
+      expect(bufferMock.toString()).toBe('68656c6c6f2c20776f726c6421');
+    });
+  });
+  describe('getOrCreateKeyPair', () => {
+    // it('called with correct arguments', async () => {
+    //   await getOrCreateKeyPair();
+    //   // expect(mockSodiumUtil.crypto_sign_seed_keypair).toBeCalledWith({});
+    //   expect(testFunc).toBeCalledWith(32, {});
+    // });
```

Comment:
you were right trying to test this.  
add check that `crypto_sign_seed_keypair` where called with correct arguments